src/com/dotmarketing/fixtask/tasks/FixTask00015FixAssetTypesInIdentifiers.java

```diff
@@ -0,0 +1,145 @@
+package com.dotmarketing.fixtask.tasks;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.dotmarketing.beans.FixAudit;
+import com.dotmarketing.beans.Inode;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.db.HibernateUtil;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.fixtask.FixTask;
+import com.dotmarketing.portlets.cmsmaintenance.ajax.FixAssetsProcessStatus;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.MaintenanceUtil;
+
+/**
+ * This task updates invalid asset-types values on identifier records by trying to match identifiers against
+ * records from corresponding asset-types and "guessing" that their existence implies correspondence to given asset-type
+ *
+ * It applies to contentlets, containers, file assets, html pages, links, and templates.
+ *
+ * @author Anibal Gomez
+ */
+public class FixTask00015FixAssetTypesInIdentifiers implements FixTask {
+
+	private HashMap<String, Integer> badDataCount = new HashMap<String, Integer>();
+	private int total = 0;
+	private String assetNames[] = { "contentlet", "containers", "file_asset", "htmlpage", "links", "template" };
+
+	public List<Map<String, Object>> executeFix() throws DotDataException,
+		DotRuntimeException {
+
+		List<Map<String, Object>> returnValue = new ArrayList<Map<String, Object>>();
+
+		Logger.info(FixTask00015FixAssetTypesInIdentifiers.class, "Beginning FixAssetTypesInIdentifiers");
+
+		if (!FixAssetsProcessStatus.getRunning()) {
+
+			FixAssetsProcessStatus.startProgress();
+			FixAssetsProcessStatus.setDescription("Task 15: Updating Invalid Identifier Asset-Types");
+			FixAssetsProcessStatus.setTotal(total);
+
+			try{
+				HibernateUtil.startTransaction();
+				DotConnect dc = new DotConnect();
+
+				for (String asset : assetNames) {
+
+					Inode.Type assetType = Inode.Type.valueOf(asset.toUpperCase());
+					final String tableName = assetType.getTableName();
+
+					Integer assetDataCount = badDataCount.get("identifier_"+asset);
+					if (assetDataCount != null && assetDataCount.intValue() > 0) {
+
+						//Update invalid entries from identifier
+						final String identifiersToUpdate = "UPDATE identifier SET asset_type = '"+ asset +"' "+
+							"WHERE asset_type <> '"+asset+"' AND EXISTS ("+
+								"SELECT * FROM "+ tableName +" WHERE identifier.id = identifier"+
+							")";
+
+						Logger.debug(MaintenanceUtil.class,"Task 15: Updating from identifier type " + asset + " : " + identifiersToUpdate);
+
+						dc.setSQL(identifiersToUpdate);
+
+						try {
+							dc.loadResult();
+						} catch (DotDataException e) {
```

Comment:
I think the try/catch may be removed. query should not fail unless there is a fatal error (such as network / database connection issue). I will remove it from FixTask00020DeleteOrphanedIdentifiers as well.